lib/src/web_ui/code_input.dart

```diff
@@ -101,6 +121,43 @@ class CodeInput {
     parenListener(missingParens);
   }
 
+  /// Determines the operation at the last open parens
+  List _currOp(String inputText, int position, [int fromLast = 1]) {
```

Comment:
It's usually best not to use a list to return multiple values of different types, since we lose type checking.

I'm okay with it if we include a detailed description of the two values (including types) in the doc comment, but I think the better way to go would be to use something like the [tuple package](https://pub.dartlang.org/packages/tuple)